cmd/jujud/agent/model/manifolds.go

```diff
@@ -386,19 +387,28 @@ func IAASManifolds(config ManifoldsConfig) dependency.Manifolds {
 // CAASManifolds returns a set of interdependent dependency manifolds that will
 // run together to administer a CAAS model, as configured.
 func CAASManifolds(config ManifoldsConfig) dependency.Manifolds {
+	agentConfig := config.Agent.CurrentConfig()
+	modelTag := agentConfig.Model()
 	manifolds := dependency.Manifolds{
 		caasBrokerTrackerName: ifResponsible(caasbroker.Manifold(caasbroker.ManifoldConfig{
 			APICallerName:          apiCallerName,
 			NewContainerBrokerFunc: config.NewContainerBrokerFunc,
 		})),
-
 		caasProvisionerName: ifNotMigrating(caasprovisioner.Manifold(
 			caasprovisioner.ManifoldConfig{
 				AgentName:     agentName,
 				APICallerName: apiCallerName,
+				BrokerName:    caasBrokerTrackerName,
 				NewWorker:     caasprovisioner.NewProvisionerWorker,
 			},
 		)),
+		modelUpgraderName: caasmodelupgrader.Manifold(caasmodelupgrader.ManifoldConfig{
```

Comment:
You seem to be going to an awful lot of trouble just to set model status and unlock the gate.

Is setting the model status from the upgrader required? If so, why?

Can you use worker/gate.AlreadyUnlocked to avoid this manifold and worker?

Reply:
Could do. But the worker also sets the model to "available". Do we want to have models without this status set? We could I guess set the status to "available" when creating the model but that seems wrong. Surely we will get upgrade steps at some point and the worker will evolve to do more than just setting the model status